### PR TITLE
dbeaver/pro#1394 show changing password dialog for users with expired…

### PR DESCRIPTION
**Pull request by LonwoLonwo**
… password

2023-03-02 17_14_07-DBeaver Ultimate 23 0 0 - _172 16 122 40_ Script-88

- [x] Please check the case from the ticket (ask me for users)
- [x] And check [password changing ](https://github.com/dbeaver/dbeaver/wiki/Change-current-user-password) in a few random databases

### CODE REVIEW

**Comment by E1izabeth** [comment]
Do we have an error saying that the password expired?
If no error is thrown before this dialog, I'd modify the message to add that password expired.

**Comment by LonwoLonwo** [comment]
> Do we have an error saying that the password expired?

It probably exists, but we can't change expired passwords like in Oracle. We can only prompt to change the password before the expiration date.

File: plugins/org.jkiss.dbeaver.ext.vertica/src/org/jkiss/dbeaver/ext/vertica/model/VerticaDataSource.java [hunk 1]
```diff
@@ -146,6 +147,14 @@ protected Connection openConnection(
                  warning = warning.getNextWarning()
             ) {
                 if (checkForPasswordWillExpireWarning(warning)) {
+                    DBAUserPasswordManager manager = getAdapter(DBAUserPasswordManager.class);
+                    if (manager != null &&
+                        DBWorkbench.getPlatformUI().confirmAction(
+                            "Change password now",
+                            "Would you like to change your password now?")
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.ext.vertica/src/org/jkiss/dbeaver/ext/vertica/model/VerticaDataSource.java]: Maybe we can write this way?
```suggestion
                            "Your password is about to expire. Would you like to change your password now?")
```
And there is no localization

Comment (LonwoLonwo) [reply plugins/org.jkiss.dbeaver.ext.vertica/src/org/jkiss/dbeaver/ext/vertica/model/VerticaDataSource.java]: This dialog is right after this one

2023-03-02 20_19_56-DBeaver Ultimate 23 0 0 - film

Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.ext.vertica/src/org/jkiss/dbeaver/ext/vertica/model/VerticaDataSource.java]: Fine. That's what I was asking about.

**Review by E1izabeth — APPROVED** [review]

**Comment by uslss** [comment]
cannot change password due to username absence

https://user-images.githubusercontent.com/31996417/225335050-5a88be35-3604-456e-b3bc-37494886aa63.mp4

**Comment by uslss** [comment]
cannot change password due to NPE

```
eclipse.buildId=unknown
java.version=11.0.18
java.vendor=Eclipse Adoptium
BootLoader constants: OS=win32, ARCH=x86_64, WS=win32, NL=en
Command-line arguments:  -os win32 -ws win32 -arch x86_64

org.jkiss.dbeaver.model
Error
Fri Mar 17 10:29:42 CET 2023
NullPointerException

java.lang.NullPointerException
	at org.jkiss.dbeaver.model.DBUtils.openMetaSession(DBUtils.java:1823)
	at org.jkiss.dbeaver.ext.vertica.model.VerticaChangeUserPasswordManager.changeUserPassword(VerticaChangeUserPasswordManager.java:41)
	at org.jkiss.dbeaver.model.access.DBAuthUtils.promptAndChangePasswordForCurrentUser(DBAuthUtils.java:64)
	at org.jkiss.dbeaver.ext.vertica.model.VerticaDataSource.openConnection(VerticaDataSource.java:156)
	at org.jkiss.dbeaver.model.impl.jdbc.JDBCExecutionContext.connect(JDBCExecutionContext.java:103)
	at org.jkiss.dbeaver.model.impl.jdbc.JDBCRemoteInstance.initializeMainContext(JDBCRemoteInstance.java:102)
	at org.jkiss.dbeaver.model.impl.jdbc.JDBCRemoteInstance.<init>(JDBCRemoteInstance.java:61)
	at org.jkiss.dbeaver.model.impl.jdbc.JDBCDataSource.initializeRemoteInstance(JDBCDataSource.java:128)
	at org.jkiss.dbeaver.ext.generic.model.GenericDataSource.<init>(GenericDataSource.java:109)
	at org.jkiss.dbeaver.ext.vertica.model.VerticaDataSource.<init>(VerticaDataSource.java:66)
	at org.jkiss.dbeaver.ext.vertica.model.VerticaMetaModel.createDataSourceImpl(VerticaMetaModel.java:57)
	at org.jkiss.dbeaver.ext.generic.GenericDataSourceProvider.openDataSource(GenericDataSourceProvider.java:106)
	at org.jkiss.dbeaver.registry.DataSourceDescriptor.openDataSource(DataSourceDescriptor.java:1191)
	at org.jkiss.dbeaver.registry.DataSourceDescriptor.connect(DataSourceDescriptor.java:1056)
	at org.jkiss.dbeaver.runtime.jobs.ConnectJob.run(ConnectJob.java:70)
	at org.jkiss.dbeaver.runtime.jobs.ConnectJob.runSync(ConnectJob.java:98)
	at org.jkiss.dbeaver.ui.actions.datasource.DataSourceHandler.connectToDataSource(DataSourceHandler.java:109)
	at org.jkiss.dbeaver.ui.actions.datasource.UIServiceConnectionsImpl.initConnection(UIServiceConnectionsImpl.java:87)
	at org.jkiss.dbeaver.model.navigator.DBNDataSource.initializeNode(DBNDataSource.java:158)
	at org.jkiss.dbeaver.model.navigator.DBNDatabaseNode.getChildren(DBNDatabaseNode.java:225)
	at org.jkiss.dbeaver.model.navigator.DBNDatabaseNode.getChildren(DBNDatabaseNode.java:1)
	at org.jkiss.dbeaver.model.navigator.DBNUtils.getNodeChildrenFiltered(DBNUtils.java:80)
	at org.jkiss.dbeaver.ui.navigator.database.load.TreeLoadService.evaluate(TreeLoadService.java:49)
	at org.jkiss.dbeaver.ui.navigator.database.load.TreeLoadService.evaluate(TreeLoadService.java:1)
	at org.jkiss.dbeaver.ui.LoadingJob.run(LoadingJob.java:88)
	at org.jkiss.dbeaver.ui.LoadingJob.run(LoadingJob.java:72)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:105)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)

```

**Comment by uslss** [comment]
previously saved user is suggested instead of current.
1. Login and save creds with user dbeaverexpire3
2. Uncheck save credentials and reconnect
3. Use dbeaverexpire4 user and don't check save credentials options
4. password change is proposed - saved user name is dbeaverexpire3


https://user-images.githubusercontent.com/31996417/226332450-eb463aaa-c1f3-4bd7-bbf7-53ddb88a66f8.mp4

**Comment by uslss** [comment]
cannot connect to database after password change in case of credentials were saved.

https://user-images.githubusercontent.com/31996417/226338293-191bbd18-3863-48ab-9ec6-d03725f6b745.mp4

**Comment by uslss** [comment]
verified

**Pull request merged**